### ISSUE TITLE
Table Schema Catalog

### ISSUE THREAD

**Issue by johanricher** [frictionlessdata/forum#5]
Following discussions with @rufuspollock, I'm proposing the creation of a new Frictionless Data product: the Table Schema Catalog.

# Context

Situation : Poor quality of open data
Question : How to improve the quality of open data?

1. Problem: Standardization of common datasets
Solution: [Table Schemas](https://frictionlessdata.io/specs/table-schema/)
Example: A schema for the names of babies born in a city in a given year (the most silly example I could think of, but [a real one nonetheless](https://scdl.opendatafrance.net/docs/schemas/prenoms.html)).

2. Problem: Checking the quality of datasets
Solution: [Goodtables](https://goodtables.io/)
Example: [Validata](https://go.validata.fr/), an adaptation of Goodtables for French open data.

3. Problem: Sharing open data standards
Solution: Schema Catalog
Example: [SCDL](https://scdl.opendatafrance.net/), Schema.data.gouv.fr, Schemas.frictionlessdata.io

# Explanation

Here's the rationale in the form of a (draft) blogpost: https://hackmd.io/@johanricher/SJmb-RazS

An extract summing up the purpose:

> For schemas to become standards, they must be easily found and usable. They must be shared. We propose to open a new chapter for Table Schemas with [Schemas.frictionlessdata.io](https://schemas.frictionlessdata.io/) as the place to catalog them.

I open this discussion with the community to brainstorm on this idea and hopefully iterate on it until it's good enough to get to work.

**Comment by rufuspollock** [comment]
@johanricher i think this is great 👍 (and something we've wanted for a while - cf https://github.com/frictionlessdata/schemas (not sure why it got deprecated)).

Could you provide a bit more detailed spec of how schemas.f15d.io would be designed e.g. its url structure e.g. do you imagine something like this?

```bash
/       # home page
/:owner/:schema-name           # human readable description of the schema  
/:owner/:schema-name.json   # json file for the schema
```

**Comment by rufuspollock** [comment]
@johanricher ping ^^^ 😉

**Comment by johanricher** [comment]
That's a good point @rufuspollock, let's talk about the sitemap and the actual workflow of this catalog!

Here are some ideas and hypothesis:

- The source of truth for each schema would stay in its owner's repo ([example](https://github.com/etalab/schema-irve)). Always name it `schema.json` and use git tags for releases.
- I would recommend [schema.org](https://schema.org/), [SPDX](https://spdx.org/licenses/) and [JSON Schema Store](http://schemastore.org/json/) as good inspirations for a successful catalog with open governance (i.e. put the data on Github and accept PRs).
- I wouldn't put usernames in the URL.
- Homepage could be schemas.frictionlessdata.io or frictionlessdata.io/schemas.

The URL structure would just be homepage/schema-name for the schema page (see below) then homepage/schema-nam}/version/schema.json to get the raw version.

Important: a /search page :mag: using faceting and filters based on the schema's metadata (see below). Finding a schema based from its country or field of application would be awesome.

Also:
- The usual /about page :nerd_face: 
- Maybe an API? :thinking:

---

A **schema page** :page_facing_up: would:

1) Lay out its profile, using [metadata from the schema itself](https://github.com/frictionlessdata/specs/blob/master/specs/patterns.md#table-schema-metadata-properties):
- Description
- Keywords / tags
- Version(s)
- Author(s)
- Image
- Country(ies) where the schema is used
- etc.

2) Offer calls to actions / tooling integration:
  - "Validate a file against this schema" (Goodtable / Validata)
  - "Create a file compliant to this schema" ([CSV-GG](https://csv-gg.etalab.studio/), [tsfaker](https://gitlab.com/healthdatahub/tsfaker/))
  - "Download a template file compliant to this schema" ([table-schema-resource-template](https://framagit.org/opendataschema/table-schema-resource-template)

3) Present the "data model" of the schema, i.e. columns ([example](https://scdl.opendatafrance.net/docs/schemas/irve.html) using [table-schema-to-markdown](https://framagit.org/opendataschema/table-schema-to-markdown))

**Comment by rufuspollock** [comment]
@johanricher this is *really* good 👏 👏 Think this is pretty much good to go. Do you want to start stubbing something out? I can boot a repo for you and make you admin!

A few suggestions / comments:

* User / org names in the url: the advantages here are 2 fold
  * Socially we have some sense of who the owner is (and the owner gets cred)
  * Technically we get to mirror github etc so we don't have to worry about name collisions
  * UX: we get an owner home page where i can browse all their schemas
* (Future) would it be useful to cache a copy of the schema
* (Future) listing users of a schema

**Comment by johanricher** [comment]
I agree that mirroring the Github/Gitlab repo URL, which people are used to, can indeed be useful and give a sense of familiarity. In some way, the schema page can be seen as just an interface on top of the repo, reading the schema and offering features to make it useful.

As for the next steps, I think I came as far as I could go alone. :smile: I can administer the repo and write user stories, stuff like that, but this would need volunteer actual developers going forward. People from my own company could help but the bandwidth is kinda limited at the moment (at least until we're able to recruit again).

No hurry on this but if we want to make progress in the next few weeks we might need to do some kind of outreach to see if other people want to contribute, what do you think?

**Comment by rufuspollock** [comment]
@johanricher I think we could make a start without devs. Could you start a spreadsheet in gdocs for a mini database of the existing schemas we know of with links etc.

**Comment by rufuspollock** [comment]
@johanricher any response here - we actually have someone who could help out here /cc @monikappv  and it would be great to get something moving.

**Comment by rufuspollock** [comment]
@johanricher pinging again 😉Do you have some time to work on this in the next few weeks?

**Comment by johanricher** [comment]
How time flies! :eyes: 

Here's what I've got for now: https://docs.google.com/spreadsheets/d/1HUHg-ITdu-aqdJyIjZXpobZZRXdCOxaxsrRLJcrcaFw/edit?usp=sharing

**Comment by rufuspollock** [comment]
@johanricher great! Could you do a sync up some time soon (e.g. this week) with @monikappv (and maybe me) to plan this out and next steps?

**Comment by johanricher** [comment]
@monikappv Can you send me propositions for times that suit you this week?

**Comment by johanricher** [comment]
@rufuspollock In the meantime, do you know lists of popular Table Schemas that I could start to add to this proto-catalog?

**Comment by monikappv** [comment]
> @monikappv Can you send me propositions for times that suit you this week?

@johanricher @rufuspollock How about tomorrow (20.02) or Friday (21.02) around 10-11 am CET?

**Comment by johanricher** [comment]
1 hour tomorrow 10am is good. :+1: Can you send me an invite? Thanks @monikappv!

**Comment by monikappv** [comment]
@johanricher sent you an invitation. Hope it's ok that I scheduled the meeting for 10:30 CET - I saw that @rufuspollock has a meeting at 10 CET. Not sure if he will be able to join us tomorrow, but just in case..

**Comment by rufuspollock** [comment]
@monikappv @johanricher can you summarize the results of your meeting and what are next steps and timeline. At the very least what about getting a blog post up detailing what we *plan* to do - even if we have not done it yet 😄

**Comment by monikappv** [comment]
@rufuspollock here are the notes from the meeting + next steps etc. https://docs.google.com/document/d/17XoirXZkYMFzSo8rs5yFQUxE8GD6KtbMKwwKjbnezy4/edit#heading=h.fuaaqygpx22k 
p.s. Thumbs up for writing a blog post although maybe a little bit too early for that.

**Comment by monikappv** [comment]
@rufuspollock summary is added to the issue description.
P.S. thumbs up for writing a blog post :+1:

**Comment by rufuspollock** [comment]
@johanricher (and cc @monikappv ) i don't think you need much syncing with me or @Irio to get this moving. Can you start on your next steps (and if there is anything you need please say).

**Comment by johanricher** [comment]
I'll have to find where I left off. Last time we talked with @monikappv was almost 2 months ago. She added the summary of that talk on the first post here. Right now, I'm not sure how I could pick up any of those "next steps".

**Comment by rufuspollock** [comment]
@johanricher could you start collecting a list of these schemas in a google doc spreadsheet and we go from that? wdyt?

Also a blog post introducing the need and idea.

**Comment by johanricher** [comment]
[Way ahead of you this time ;)](https://github.com/frictionlessdata/forum/issues/5#issuecomment-586615992)
> Here's what I've got for now: https://docs.google.com/spreadsheets/d/1HUHg-ITdu-aqdJyIjZXpobZZRXdCOxaxsrRLJcrcaFw/edit?usp=sharing

**Comment by johanricher** [comment]
As for the blogpost, [my first draft](https://hackmd.io/@johanricher/SJmb-RazS) could be reworked of course, but it would depend on the angle. Where would it be posted? Would it be to present a work in progress? Would we have something else to show for it by then? or would the intent be to do some kind of outreach to see if other people want to contribute, as suggested before?

**Comment by rufuspollock** [comment]
> As for the blogpost, my first draft could be reworked of course, but it would depend on the angle. Where would it be posted? Would it be to present a work in progress? Would we have something else to show for it by then? or would the intent be to do some kind of outreach to see if other people want to contribute, as suggested before?

@johanricher the intent of the blog post would be to announce the idea and that we are starting to work on it. I think getting it up asap on the new site would be great 😄 - if you  want someone to review @lauragift21 can do that and then we can get a PR in.

**Comment by johanricher** [comment]
Great, thanks!

Hi @lauragift21 Would you like to read [this draft blog post](https://hackmd.io/@johanricher/SJmb-RazS) and give me some quick notes? We can use whatever medium you'd like (Google Docs, Github...) to iterate and improve it until it's ready to publish on the Frictionless Data Blog.

**Comment by lauragift21** [comment]
Hi @johanricher Sure happy to take a look and have a review sent by Monday.

**Comment by lauragift21** [comment]
The post looks good to me. I suggested a few grammatical edits in the document you can check those out. @johanricher

**Comment by johanricher** [comment]
Thanks for your speed of response @lauragift21!

@rufuspollock What do you think about adding an invitation to give feedback somewhere? A new Github issue or thread on the OKFN forum, perhaps?

**Comment by rufuspollock** [comment]
@johanricher sounds good in terms of adding invitation for feedback. I
think the easiest would be an "open" google doc or to post comment here in
the github forum.

**Comment by johanricher** [comment]
Sorry, I meant when the post will be published, what do we propose to people that are interested to get feedback from them?

As for the blog post itself, corrections and/or comments are already possible [on the draft here](https://hackmd.io/@johanricher/SJmb-RazS).

**Comment by rufuspollock** [comment]
@johanricher yes i meant for when blog post published - a google doc would be fine or a link to here.

**Comment by rufuspollock** [comment]
@lauragift21 can you take the lead on liasing with @johanricher and getting this published.

Aside: I think this project could be worth a FEP - see frictionlessdata/project#563

**Comment by lauragift21** [comment]
Sure @rufuspollock.  

Hey, @johanricher I think adding a link to this issue is worth a try I like that idea better than Google Docs because it's public. I can publish this on the blog once you confirm it is good to go.

**Comment by johanricher** [comment]
If you think it's good enough for publication, then I say let's do it! :)

**Comment by lauragift21** [comment]
Alright, I'll have it published tomorrow. :)

**Comment by lauragift21** [comment]
Hey @johanricher. We now have your post published on the blog. Let me know if you want to use a custom image and I'll have that updated. https://frictionlessdata.io/blog/2020/04/23/table-schema-catologue/

**Comment by johanricher** [comment]
Awesome, thank you for the publication! I agree that some images would be an improvement but I don't have anything to offer for now.

Can you add a phrase at the end, like an invitation to give feedback on this issue here? Also, there's a minor typo in the slug ("catolog").

**Comment by lauragift21** [comment]
Nice catch. I'll push a fix.

**Comment by lauragift21** [comment]
Hello @johanricher 👋 

Do you think you can let us know a time when you're free so I can schedule a meeting with @monikappv regarding the next steps for this project?

**Comment by johanricher** [comment]
Hi! sure, can we schedule something for next week? Anytime Wednesday to Friday is fine by me. You have my email.

To be clear, do you intend to talk about the list of "next steps" already made by @monikappv (see above) or scratch that completely?

**Comment by lauragift21** [comment]
Thank @johanricher I'll schedule a meeting. I think it'll be helpful to have a follow-up from the meeting with @monikappv and set actionable steps to help us get started.
